server/src/test/java/org/elasticsearch/cluster/MinimumMasterNodesIT.java

```diff
@@ -157,12 +184,23 @@ public void testSimpleMinimumMasterNodes() throws Exception {
 
         ensureGreen();
 
-        logger.info("--> verify we the data back after cluster reform");
+        logger.info("--> verify we get the data back after cluster reform");
         for (int i = 0; i < 10; i++) {
             assertHitCount(client().prepareSearch().setSize(0).setQuery(QueryBuilders.matchAllQuery()).execute().actionGet(), 100);
         }
 
-        internalCluster().stopRandomNonMasterNode();
+        logger.info("--> clearing voting config exclusions");
+        ClearVotingConfigExclusionsRequest clearRequest = new ClearVotingConfigExclusionsRequest();
+        clearRequest.setWaitForRemoval(false);
+        client().execute(ClearVotingConfigExclusionsAction.INSTANCE, clearRequest).get();
+
+        masterNode = internalCluster().getMasterName();
+        otherNode = node1Name.equals(masterNode) ? node2Name : node1Name;
+        logger.info("--> add voting config exclusion for master node, to be sure it's not elected");
+        client().execute(AddVotingConfigExclusionsAction.INSTANCE, new AddVotingConfigExclusionsRequest(new String[]{masterNode})).get();
+        logger.info("--> stop non-master node, no master block should appear");
```

Comment:
should we clear voting exclusions after shutting the node down?

Reply:
Since we will have a "no master block", it's not possible to clear voting exclusions.

Reply:
ah my bad